### ISSUE TITLE
I think the wrong value is being passed to strerror here

### ISSUE THREAD

**Issue by scott-bossanova** [aarond10/https_dns_proxy#78]
At [line 53-55 of dns_server.c](https://github.com/aarond10/https_dns_proxy/blob/de2321f0f255b9317488ee3441697d999130db0c/src/dns_server.c#L53-L55) the result of `bind()` is passed to `strerror`. The manual for bind says "_Upon successful completion, bind() shall return 0; otherwise, −1 shall be returned and errno set to indicate the error._". The manual for strerror indicates it returns "_a string that describes the error code passed in the argument errnum_".

I'm not very experienced in C, but based on what I've read, I think lines 53-55 should read

```c
  if ((res = bind(sock, ai->ai_addr, ai->ai_addrlen)) < 0) {
    FLOG("Error binding %s:%d: %s", listen_addr, listen_port, strerror(errno));
  }
```
I discovered this because when running the example command in the readme for setting DNS to cloudflare on my openwrt installation, and received the following error:
```
dns_server.c:54 Error binding 127.0.0.1:5053: No error information
```

**Comment by dscottboggs** [comment]
Oops, wrong github acount

**Comment by aarond10** [comment]
Thanks! Good catch. Fixed in a8dbe92
